src/torch_models.py

```diff
@@ -476,6 +504,43 @@ def _predict(self, x: Array) -> Array:
         sample_idx = torch.argmax(scores)
         return sample_ys[sample_idx]
 
+    def _predict_derivative_free(self, x: Array) -> Array:
+        # Reference: https://arxiv.org/pdf/2109.00137.pdf (Algorithm 1).
+        # This method reportedly works well in up to 5 dimensions.
+        # Since we are using torch for random sampling, and since we want
```

Comment:
this is shady; why do we need to reseed? we already seed at the start. is it because training could use up an unknown amount of seeding? if so, i'd rather we just call torch.manual_seed() at the end of train_pytorch_model, right around where we call `model.eval()`

Reply:
discussed on slack. it's not just load, but also option model vs execution